data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3ScanPartitionCreationSupplier.java

```diff
@@ -174,14 +176,22 @@ private Instant getMostRecentLastModifiedTimestamp(final ListObjectsV2Response l
     }
 
     private boolean shouldScanBeSkipped(final Map<String, Object> globalStateMap) {
+
         if (Objects.isNull(schedulingOptions) && hasAlreadyBeenScanned(globalStateMap)) {
-            LOG.info("Skipping scan because the buckets have already been scanned once");
+
+            if (!(Boolean) globalStateMap.get(SINGLE_SCAN_COMPLETE)) {
+                LOG.info("Single S3 scan has already been completed");
+                globalStateMap.put(SINGLE_SCAN_COMPLETE, true);
```

Comment:
What's the difference between `SINGLE_SCAN_COMPLETE` and `SCAN_COUNT`? can we just use SCAN_COUNT to determine if scan is complete?

Reply:
SCAN_COUNT could block scanning since the global state doesn't go away. so it's better to keep it separate for this log